src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -495,7 +497,7 @@ private Object externalValueToInternalValue(GraphqlFieldVisibility fieldVisibili
             Object returnValue =
                     externalValueToInternalValue(fieldVisibility, unwrapOne(graphQLType), value);
             if (returnValue == null) {
-                throw new NonNullableValueCoercedAsNullException("", emptyList(), graphQLType);
+                throw new NonNullableValueCoercedAsNullException(emptyList(), graphQLType);
```

Comment:
The problem I see here is `emptyList` is not much better than `""` - that is there is no real info available.

Reply:
I mean its better than it was - but is it as good as it could be?

I guess we would have to "track" fields and types as we descending the ValueResolver to get excellent names in the exception

Reply:
I am happy to merge this as progress over perfection

Reply:
I would rather remove the `emptyList()` path

Reply:
I have another PR - https://github.com/graphql-java/graphql-java/pull/2773 - that means this is less likely in terms of conditions where this will happen - still possible but less likely